client/resource_group/controller/controller.go

```diff
@@ -297,6 +320,10 @@ func (c *ResourceGroupsController) sendTokenBucketRequests(ctx context.Context,
 		Requests:              requests,
 		TargetRequestPeriodMs: uint64(defaultTargetPeriod / time.Millisecond),
 	}
+	if c.run.responseDeadline == nil {
+		c.run.responseDeadline = time.NewTimer(time.Second)
```

Comment:
I suggest using `Stop()` and `Rest()` here to re-use the timer as much as possible rather than creating a new one each time.